p2p/switch.go

```diff
@@ -328,6 +328,11 @@ func (sw *Switch) reconnectToPeer(addr *NetAddress) {
 			return
 		}
 
+		if sw.IsDialingOrExistingAddress(addr) {
```

Comment:
Thanks. These changes are partial fixes for https://github.com/tendermint/tendermint/issues/2716.

Note the races are still possible, but this should help significantly. We'll likely have to rearchitect the code a bit to solve this fully.

Thanks!